File: LeetCode (en comments)/88. Merge Sorted Array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        k = m + n - 1

        while n-1 >= 0:
            if m-1 >= 0 and nums1[m-1] > nums2[n-1]:
                nums1[k] = nums1[m-1]
                m -= 1
            else:
                nums1[k] = nums2[n-1]
                n -= 1
            k -= 1
        return nums1

File: LeetCode (en comments)/88. Merge Sorted Array.c
void merge(int* nums1, int nums1Size, int m, int* nums2, int nums2Size, int n) {
    int k = m + n - 1;

    while (n > 0) {
        if (m > 0 && nums1[m - 1] > nums2[n - 1]) {
            nums1[k] = nums1[m - 1];
            m--;
        } else {
            nums1[k] = nums2[n - 1];
            n--;
        }
        k--;
    }
}